CMakeLists.txt

```diff
@@ -118,7 +118,15 @@ else ()
     -Wuninitialized
   )
 
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wold-style-cast")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wold-style-cast")
+
+  if(NOT CYGWIN)
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+  else()
+      # On Cygwin, POSIX extensions are not visible with std=c++11 
+      # because it defines __STRICT_ANSI__.
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11")
```

Comment:
Would it be better to define _POSIX_SOURCE or something like that to signal that we want POSIX rather than asking for GNU extensions?  I also don't think its a great idea to use different `-std` on different platforms.

Reply:
Yup, there's room for improvement. Currently, the only POSIX consumers are https://github.com/WebAssembly/wabt/blob/4ba97c116d1417b2fb516299a37965820b93ced3/src/color.cc#L69 AND gtest. For the `color.cc` we'd just have `_POSIX_C_SOURCE` on it but I'm not sure how do we want to deal with gtest... 

So I choose to limit `gnu++11` to Cygwin and leave it as-is on the other (major) implementations such as glibc(Linux) or BSD libc.

Reply:
I think it would make sense to define _POSIX_C_SOURCE for the whole project (assuming that works) than turn on GNU extensions.

Reply:
Okay, I'll try. I found gtest upstream also have a proposed patch https://github.com/google/googletest/pull/2653/files so we should be able to do the same here.

The only culprit seems `strcasecmp` which is also POSIX. I'll chase why Cygwin does not export it under `_POSIX_C_SOURCE=200809L`.

Reply:
Filed: #1333